File: PODM/SW/pod-manager/podm-business-entities/src/main/java/com/intel/podm/business/entities/redfish/properties/GraphicalConsole.java
/*
 * Copyright (c) 2016 Intel Corporation
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.intel.podm.business.entities.redfish.properties;

import com.intel.podm.business.entities.base.DomainObject;
import com.intel.podm.business.entities.base.DomainObjectProperty;
import com.intel.podm.common.types.GraphicalConnectType;
import com.intel.podm.common.types.helpers.EnumeratedTypeListHolder;

import javax.enterprise.context.Dependent;
import javax.transaction.Transactional;
import java.util.List;

import static com.intel.podm.business.entities.base.DomainObjectProperties.booleanProperty;
import static com.intel.podm.business.entities.base.DomainObjectProperties.enumeratedTypeListProperty;
import static com.intel.podm.business.entities.base.DomainObjectProperties.integerProperty;
import static com.intel.podm.common.types.helpers.EnumeratedTypeListHolder.enumeratedTypeListHolder;
import static com.intel.podm.common.types.helpers.EnumeratedTypeListHolder.toList;
import static javax.transaction.Transactional.TxType.MANDATORY;

@Dependent
@Transactional(MANDATORY)
public class GraphicalConsole extends DomainObject {
    public static final DomainObjectProperty<Boolean> SERVICE_ENABLED = booleanProperty("serviceEnabled");
    public static final DomainObjectProperty<Integer> MAX_CONCURRENT_SESSIONS = integerProperty("maxConcurrentSessions");
    public static final DomainObjectProperty<EnumeratedTypeListHolder<GraphicalConnectType>> CONNECT_TYPES_SUPPORTED =
            enumeratedTypeListProperty("connectTypesSupported", GraphicalConnectType.class);

    public Boolean isServiceEnabled() {
        return getProperty(SERVICE_ENABLED);
    }

    public void setServiceEnabled(Boolean serviceEnabled) {
        setProperty(SERVICE_ENABLED, serviceEnabled);
    }

    public Integer getMaxConcurrentSessions() {
        return getProperty(MAX_CONCURRENT_SESSIONS);
    }

    public void setMaxConcurrentSessions(Integer maxConcurrentSessions) {
        setProperty(MAX_CONCURRENT_SESSIONS, maxConcurrentSessions);
    }

    public List<GraphicalConnectType> getConnectTypesSupported() {
        return toList(GraphicalConnectType.class, getProperty(CONNECT_TYPES_SUPPORTED));
    }

    public void setConnectTypesSupported(List<GraphicalConnectType> connectTypesSupported) {
        setProperty(CONNECT_TYPES_SUPPORTED, enumeratedTypeListHolder(GraphicalConnectType.class, connectTypesSupported));
    }
}

File: PSME/common/agent-framework/src/module/model/attributes/graphical_console.cpp
/*!
 * @section LICENSE
 *
 * @copyright
 * Copyright (c) 2015-2019 Intel Corporation
 *
 * @copyright
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * @copyright
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * @copyright
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * @section DESCRIPTION
 *
 * */

#include "agent-framework/module/model/attributes/graphical_console.hpp"
#include "agent-framework/module/constants/common.hpp"
#include "json-wrapper/json-wrapper.hpp"

using namespace agent_framework::model;
using namespace agent_framework::model::attribute;

json::Json GraphicalConsole::to_json() const {
    json::Json json = json::Json();
    json[literals::GraphicalConsole::ENABLED] = get_enabled();
    json[literals::GraphicalConsole::MAX_SESSIONS] = get_max_sessions();
    json[literals::GraphicalConsole::TYPES_SUPPORTED] = get_types_supported().to_json();
    return json;
}

GraphicalConsole GraphicalConsole::from_json(const json::Json& json) {
    GraphicalConsole console;
    console.set_enabled(json[literals::GraphicalConsole::ENABLED]);
    console.set_max_sessions(json[literals::GraphicalConsole::MAX_SESSIONS]);
    console.set_types_supported(SupportedConnections::from_json(
        json[literals::GraphicalConsole::TYPES_SUPPORTED]));
    return console;
}
